pkg/controller/tsp/tsp_controller.go

```diff
@@ -45,6 +49,12 @@ func NewController(
 		Recorder:     recorder,
 		UpdatePeriod: updatePeriod,
 		predictors:   predictors,
+		metric: prometheus.NewDesc(
```

Comment:
metrics definition and controller structure should be decouple per my understanding.

Reply:
ok